jme3-plugins/src/gltf/java/com/jme3/scene/plugins/gltf/LightsPunctualExtensionLoader.java

```diff
@@ -0,0 +1,295 @@
+        color = lumensToColor(color, intensity);
+        float range = obj.has("range") ? obj.get("range").getAsFloat() : Float.POSITIVE_INFINITY;
+
+        PointLight pointLight = new PointLight();
+        pointLight.setName(name);
+        pointLight.setColor(color);
+        pointLight.setRadius(range);
+
+        return pointLight;
+    }
+
+    /**
+     * Attach a light at the given index to the given parent node,
+     * and the control for the light to the given node.
+     * @param parent The node to attach the light to
+     * @param node The node to attach the light control to
+     * @param lightIndex The index of the light
+     */
+    private void addLight(Node parent, Node node, int lightIndex) {
+        if (lightDefinitions.containsKey(lightIndex)) {
+            Light light = lightDefinitions.get(lightIndex);
+            parent.addLight(light);
+            LightControl control = new LightControl(light);
+            node.addControl(control);
+        } else {
+            throw new AssetLoadException("KHR_lights_punctual extension accessed undefined light at index " + lightIndex);
+        }
+    }
+
+    /**
+     * Convert a floating point lumens value into a color that
+     * represents both color and brightness of the light.
+     *
+     * @param color The base color of the light
+     * @param lumens The lumnes value to convert to a color
+     * @return A color representing the intensity of the given lumens encoded into the given color
+     */
+    private ColorRGBA lumensToColor(ColorRGBA color, float lumens) {
+        ColorRGBA brightnessModifier = lumensToColor(lumens);
+        return color.mult(brightnessModifier);
+    }
+
+    /**
+     * Convert a floating point lumens value into a grayscale color that
+     * represents a brightness.
+     *
+     * @param lumens The lumnes value to convert to a color
+     * @return A color representing the intensity of the given lumens
+     */
+    private ColorRGBA lumensToColor(float lumens) {
+        /*
+        Taken from /Common/ShaderLib/Hdr.glsllib
+        vec4 HDR_EncodeLum(in float lum){
+        float Le = 2.0 * log2(lum + epsilon) + 127.0;
+        vec4 result = vec4(0.0);
+        result.a = fract(Le);
+        result.rgb = vec3((Le - (floor(result.a * 255.0)) / 255.0) / 255.0);
+        return result;
+         */
+        float epsilon = 0.0001f;
+
+        double Le = 2f * Math.log(lumens * epsilon) / Math.log(2) + 127.0;
+        ColorRGBA color = new ColorRGBA();
+        color.a = (float) (Le - Math.floor(Le)); //Get fractional part
+        float val = (float) ((Le - (Math.floor(color.a * 255.0)) / 255.0) / 255.0);
+        color.r = val;
+        color.g = val;
+        color.b = val;
+
+        return color;
+    }
+
+    /**
+     * A bean to contain the relation between a node and a light index
+     */
+    private static class NodeNeedingLight {
```

Comment:
Does this private class need public methods?

Reply:
No, good catch. I will correct when you are finished with your review.